unity/Assets/Scripts/AgentManager.cs

```diff
@@ -1122,21 +1113,19 @@ public class OptionalVector3 {
         this.renderImage = controlCommand.renderImage;
         this.activeAgentId = controlCommand.agentId;
 
-        var args = new DynamicServerAction(
-            new JObject(
-                controlCommand.jObject.Properties().Where(p => !injectedArguments.Contains(p.Name))
-            )
-        );
-
         if (agentManagerActions.Contains(controlCommand.action)) {
             // let's look in this class for the action
-            this.activeAgent().ProcessControlCommand(controlCommand: args, target: this);
+            this.activeAgent().ProcessControlCommand(controlCommand: controlCommand, target: this);
```

Comment:
Now that `action` is included, I would do the filtering in `BaseFPSAgent`, `ProcessControlCommand` right before the dispatch, and create a new args object. Like make sure that Dispatch is called always with the right arguments, I think that will simplify logic and reduce future bugs.